src/EnergyPlus/OutputProcessor.cc

```diff
@@ -1689,6 +1697,22 @@ namespace OutputProcessor {
 			VarsOnCustomMeter = 0;
 			MaxVarsOnCustomMeter = 1000;
 			NumVarsOnCustomMeter = 0;
+			// check if any fields reference another Meter:Custom
+			int found = 0;
+			for ( fldIndex = 4; fldIndex <= NumAlpha; fldIndex += 2 ) {
+				for ( auto nameOfMeterCustom : namesOfMeterCustom ) {
```

Comment:
With a change to using set, I would rewrite all this code as follows...

```cpp
bool referenceMeterCustom = false;
for ( fldIndex = 4; fldIndex <= NumAlpha; fldIndex += 2 ) {
  if ( namesOfMeterCustom.find( MakeUPPERCase( cAlphaArgs( fldIndex ) ) ) != namesOfMeterCustom.end() ) {
    referenceMeterCustom = true;
    break;
  }
}
if ( referenceMeterCustom ) {
  ShowWarningError (cCurrentModuleObject + "=\"" + cAlphaArgs (1) + "\", contains a reference to another " + cCurrentModuleObject + "in field: " + cAlphaFieldNames (found) + "=\"" + cAlphaArgs (found) + "\".");
  continue;
}
```

With these changes, this goes from an O( n * m ) algorithm to O( n * log( m ) ) for set or O( n ) for unordered_set aka faster.